From: Michael Froman <mfroman@mozilla.com>
Date: Tue, 13 May 2025 22:39:00 +0000
Subject: Bug 1966238 - Cherry-pick upstream libwebrtc commit f1b18e0b4b
 r?pehrsons!

Upstream commit: https://webrtc.googlesource.com/src/+/f1b18e0b4b723223de5efef859bfa6feef19875a
       DesktoCapturer: Stop invoking ScreenCapturerSck code on iOS

       Follow-up to r44426, which made DesktopCapturer::CreateGenericCapturer()
       call CreateGenericCapturerSck() gated on WEBRTC_MAC.

       WEBRTC_MAC is also defined for iOS (and tvOS) builds, but the actual
       implementation is Mac-specific, leading to undefined symbol errors in
       Blingium debug builds.

       Bug: webrtc:367915807, chromium:391914246, chromium:40254930
       Change-Id: I2407297bdcf93b39a51deddac546420a7f5bffcc
       Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/388261
       Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
       Commit-Queue: Raphael Kubo da Costa <kubo@igalia.com>
       Reviewed-by: Alexander Cooper <alcooper@chromium.org>
       Cr-Commit-Position: refs/heads/main@{#44453}

Differential Revision: https://phabricator.services.mozilla.com/D249165
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/c8dcb907bdbae7ab2b2d6d343403c53882465a88
---
 AUTHORS                                     | 2 +-
 modules/desktop_capture/desktop_capturer.cc | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/AUTHORS b/AUTHORS
index cad8221ad2..0599f49da1 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -89,7 +89,6 @@ Mallikarjuna Rao V <vm.arjun@samsung.com>
 Manish Jethani <manish.jethani@gmail.com>
 Martin Storsjo <martin@martin.st>
 Matthias Liebig <matthias.gcode@gmail.com>
-Maksim Sisov <msisov@igalia.com>
 Maxim Pavlov <pavllovmax@gmail.com>
 Maxim Potapov <vopatop.skam@gmail.com>
 Michael Iedema <michael@kapsulate.com>
@@ -178,6 +177,7 @@ Google Inc. <*@google.com>
 Highfive, Inc. <*@highfive.com>
 Hopin Ltd. <*@hopin.to>
 HyperConnect Inc. <*@hpcnt.com>
+Igalia S.L. <*@igalia.com>
 Intel Corporation <*@intel.com>
 LG Electronics, Inc. <*@lge.com>
 Life On Air Inc. <*@lifeonair.com>
diff --git a/modules/desktop_capture/desktop_capturer.cc b/modules/desktop_capture/desktop_capturer.cc
index 45c13751aa..94605f409c 100644
--- a/modules/desktop_capture/desktop_capturer.cc
+++ b/modules/desktop_capture/desktop_capturer.cc
@@ -30,7 +30,7 @@
 #include "modules/desktop_capture/linux/wayland/base_capturer_pipewire.h"
 #endif
 
-#if defined(WEBRTC_MAC)
+#if defined(WEBRTC_MAC) && !defined(WEBRTC_IOS)
 #include "modules/desktop_capture/mac/screen_capturer_sck.h"
 #endif
 
@@ -121,7 +121,7 @@ std::unique_ptr<DesktopCapturer> DesktopCapturer::CreateGenericCapturer(
     capturer = std::make_unique<BaseCapturerPipeWire>(
         options, CaptureType::kAnyScreenContent);
   }
-#elif defined(WEBRTC_MAC)
+#elif defined(WEBRTC_MAC) && !defined(WEBRTC_IOS)
   capturer = CreateGenericCapturerSck(options);
 #endif
 
